## Задача 4. 
### Даны два списка: Дата покупки и Суммы покупок по датам

- ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']
- [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]

### 1. Найдите, какая выручка у компании в ноябре.  Используйте list comprehensions.

In [3]:
date = ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', 
        '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', 
        '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', 
        '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', 
        '2021-10-26', '2021-12-09']
count = [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 
         2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]


In [4]:
def revenue_nov(date1: list, count1: list) -> list:
    
    new_rev = []
    
    for i in range(len(date1)-1):
        if '-11-' in date1[i]:
            new_rev.append(count1[i])
            
    return new_rev

print(sum(revenue_nov(date, count)))

25098


In [5]:
print(sum([count[i] for i in range(len(date)-1) if "-11" in date[i]]))

25098


### 2. Найдите выручку компании в зависимости от месяца. Напишите функцию, которая на вход принимает списки с датами и с выручкой, а на выходе вернет словарь, где ключи - это месяцы, а значения - это выручка.
Примеры/Тесты:

        <finction_name>(dates, prices) -> {'09': 25647, '12': 45452, '10': 28645, '11': 25098}

*Примечание:* 
- Для извлечения месяца вы можете воспользоваться пакетом datetime или извлекать его как строку
- Для работы "синхронно" с двумя последовательностями вы можете использовать индекс или функцию zip. Попробуйте оба варианта.
- Используйте аннотирование типов при определении функции

In [7]:
def revenue (date: list, count: list) -> dict:
    revenue = {}
    for i in range(len(date)):
        
        if date[i].split('-')[1] in revenue.keys():
            revenue[date[i].split('-')[1]] += count[i]
        else:
            revenue[date[i].split('-')[1]] = count[i]
            
    return revenue
print(revenue(date, count))


{'09': 25647, '12': 45452, '10': 28645, '11': 25098}


In [9]:
print(sum(list((d[1]*10 for d in zip(date,count)if d[0].split('-')[1] == '11'))))

250980
